app/tiebout.hoon

```diff
@@ -0,0 +1,236 @@
+::
+::  state:
+::
+|_  [bol=bowl:gall sta=state]
+::
+::  +this: app core subject
+::
+++  this  .
+::
+::  +prep: set up app state, upgrade app state
+::
+++  prep
+  |=  old=(unit state)
+  ^-  (quip move _this)
+  ?~  old
+    :-  ~
+    %=  this
+      king.sta  ~dabben-larbet
+      baseurl.sta  'https://api.push.apple.com/3/device/'
+    ==
+  ?-  -.u.old
+    %0
+      [~ this(sta u.old)]
+  ==
+::
+::  +coup: receive acknowledgement for poke, print error if it failed
+::
+++  coup
+  |=  [wir=wire err=(unit tang)]
+  ^-  (quip move _this)
+  ?~  err
+    [~ this]
+  (mean u.err)
+::
+::  +poke-noun: receive debugging actions
+::
+++  poke-noun
+  |=  act=action
+  ^-  (quip move _this)
+  (poke-tiebout-action act)
+::
+::  +poke-tiebout-action: main action handler
+::
+++  poke-tiebout-action
+  |=  act=action
+  ^-  (quip move _this)
+  ?-  -.act
+    $king        (set-king +.act)
+    $token       (set-token +.act)
+    $baseurl     (set-baseurl +.act)
+    $add-circle  (add-circle +.act)
+    $del-circle  (del-circle +.act)
+    $notify      (send-notify +.act)
+  ==
+::
+::  +add-circle: add circle and subscribe for updates
+::
+++  add-circle
+  |=  nom=name:hall
+  ^-  (quip move _this)
+  :_  this(circles.sta (~(put by circles.sta) nom 0))
+  [ost.bol %peer /our/[nom] [our.bol %hall] /circle/[nom]/config/grams]~
+::
+::  +del-circle: delete circle and unsubscribe from updates
+::
+++  del-circle
+  |=  nom=name:hall
+  ^-  (quip move _this)
+  :_  this(circles.sta (~(del by circles.sta) nom))
+  [ost.bol %pull /our/[nom] [our.bol %hall] ~]~
+::
+::  +set-king: set king @p
+::
+++  set-king
+  |=  kng=@p
+  ^-  (quip move _this)
+  [~ this(king.sta kng)]
+::
+::  +set-token: set iOS device token @t
+::
+++  set-token
+  |=  tok=@t
+  ^-  (quip move _this)
+  [~ this(token.sta tok)]
+::
+::  +set-baseurl: set base url @t
+::
+++  set-baseurl
+  |=  burl=@t
+  ^-  (quip move _this)
+  [~ this(baseurl.sta burl)]
+::
+::  +send-notify: if king, send hiss. if not, do nothing.
+::
+++  send-notify
+  |=  not=notification
+  ^-  (quip move _this)
+  ?:  =(king.sta our.bol)
+    :_  this
+    [ost.bol %hiss /request [~ ~] %httr %hiss (create-apns-request not)]~
+  [~ this]
+::
+::  +diff-hall-prize: receive new circle data
+::
+++  diff-hall-prize
+  |=  [wir=wire piz=prize:hall]
+  ^-  (quip move _this)
+  ?+  wir
+    (mean [leaf+"invalid wire for diff: {(spud wir)}"]~)
+  ::
+  ::  %our: set config of circle and iterate through messages, sending
+  ::  notifications for all messages where number is higher than our last-read
+  ::
+      {%our @ @}
+    ?>  ?=(%circle -.piz)
+    =/  nom/name:hall  i.t.wir
+    =/  red/@ud  red.loc.cos.piz
+    [~ this(circles.sta (~(put by circles.sta) nom red))]
+  ==
+::
+::  +reap: recieve acknowledgement for peer
+::
+++  reap
+  |=  [wir=wire err=(unit tang)]
+  ^-  (quip move _this)
+  ?~  err
+    [~ this]
+  ?~  wir
+    (mean [leaf+"invalid wire for diff: {(spud wir)}"]~)
+  ?+  i.wir
```

Comment:
I thought I left a comment about this in the initial review but can't find it, so maybe it got eaten? You already do a saner `?+` in the `++diff-hall-*` cases, you'll want to apply that here and in `++quit` too.

(Also indenting for this particular one is borked.)